lib/inspec/cli.rb

```diff
@@ -163,11 +163,9 @@ def shell_func
     o[:logger].level = get_log_level(o.log_level)
     if o[:command].nil?
       runner = Inspec::Runner.new(o)
-      return Inspec::Shell.new(runner).start
+      return Inspec::Shell.new(runner, o).start
     else
-      res = run_command(o)
```

Comment:
run_command should already print the json output.  It's no longer using a mock runner - it's doing the real rspec thing which handles --formatter on its own.